jaeger-core/src/main/java/com/uber/jaeger/propagation/B3TextMapCodec.java

```diff
@@ -47,44 +49,52 @@
 
   @Override
   public void inject(SpanContext spanContext, TextMap carrier) {
-    carrier.put(TRACE_ID_NAME, HexCodec.toLowerHex(spanContext.getTraceId()));
-    if (spanContext.getParentId() != 0L) { // Conventionally, parent id == 0 means the root span
-      carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
-    }
-    carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
-    carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
-    if (spanContext.isDebug()) {
-      carrier.put(FLAGS_NAME, "1");
+    try {
+      carrier.put(TRACE_ID_NAME, HexCodec.toLowerHex(spanContext.getTraceId()));
+      if (spanContext.getParentId() != 0L) { // Conventionally, parent id == 0 means the root span
+        carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
+      }
+      carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
+      carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
+      if (spanContext.isDebug()) {
+        carrier.put(FLAGS_NAME, "1");
+      }
+    } catch (RuntimeException ex) {
+      log.error("Error when injecting SpanContext into carrier. Handling gracefully.", ex);
     }
   }
 
   @Override
   public SpanContext extract(TextMap carrier) {
-    Long traceId = null;
-    Long spanId = null;
-    Long parentId = 0L; // Conventionally, parent id == 0 means the root span
-    byte flags = 0;
-    for (Map.Entry<String, String> entry : carrier) {
-      if (entry.getKey().equalsIgnoreCase(SAMPLED_NAME)) {
-        String value = entry.getValue();
-        if ("1".equals(value) || "true".equalsIgnoreCase(value)) {
-          flags |= SAMPLED_FLAG;
-        }
-      } else if (entry.getKey().equalsIgnoreCase(TRACE_ID_NAME)) {
-        traceId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
-      } else if (entry.getKey().equalsIgnoreCase(PARENT_SPAN_ID_NAME)) {
-        parentId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
-      } else if (entry.getKey().equalsIgnoreCase(SPAN_ID_NAME)) {
-        spanId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
-      } else if (entry.getKey().equalsIgnoreCase(FLAGS_NAME)) {
-        if (entry.getValue().equals("1")) {
-          flags |= DEBUG_FLAG;
+    try {
+      Long traceId = null;
+      Long spanId = null;
+      Long parentId = 0L; // Conventionally, parent id == 0 means the root span
+      byte flags = 0;
+      for (Map.Entry<String, String> entry : carrier) {
+        if (entry.getKey().equalsIgnoreCase(SAMPLED_NAME)) {
+          String value = entry.getValue();
+          if ("1".equals(value) || "true".equalsIgnoreCase(value)) {
+            flags |= SAMPLED_FLAG;
+          }
+        } else if (entry.getKey().equalsIgnoreCase(TRACE_ID_NAME)) {
```

Comment:
not related to this PR but there can be NPE if the key is null, better to write it as `TRACE_ID_NAME.equals()`

Reply:
yep, I noticed that but didn't want to make uber(haha) request ;)